utils/models_setup.py

```diff
@@ -30,7 +30,24 @@ def get_device(use_gpu=True, rank_per_model=1):
         raise ValueError("Exactly 1 rank per device currently supported")
 
     print("Using GPU")
-    device_name = "cuda:" + str(world_rank)
+    ## We need to ge a local rank if there are multiple GPUs available.
+    localrank = 0
+    if torch.cuda.device_count() > 1:
+        if os.getenv("OMPI_COMM_WORLD_LOCAL_RANK"):
+            ## Summit
+            localrank = int(os.environ["OMPI_COMM_WORLD_LOCAL_RANK"])
+        elif os.getenv("SLURM_LOCALID"):
+            ## CADES
+            localrank = int(os.environ["SLURM_LOCALID"])
+
+        if localrank >= torch.cuda.device_count():
+            print(
+                "WARN: localrank is greater than the available device count - %d %d"
+                % (localrank, torch.cuda.device_count())
+            )
+
+    device_name = "cuda:" + str(localrank)
```

Comment:
Should this work for either method of assigning resource sets on Summit?

Reply:
I tested the following two cases working:
```
jsrun -n12 -a1 -g1 -c1 -r6 
jsrun -n2 -a6 -g6 -c6 -r1
```

However, currently it will break with the following case:
```
jsrun -n4 -a3 -g3 -c3 -r2
```
I have no clear solution/idea yet.

Reply:
I think the first two cases working is all we need